internal/zinc-core/src/main/scala/sbt/internal/inc/Relations.scala

```diff
@@ -605,7 +605,7 @@ private class MRelationsNameHashing(
       internalDependencies = internalDependencies,
       externalDependencies = externalDependencies,
       classes,
-      names = names + (className, name),
+      names = names + (className, name.replaceAllLiterally("\n", "")),
```

Comment:
Can you hoist up this expression into a function that is invoked in a hoisted up value?
  
Also, this patch makes both a name with a new line and a name without a new line to have the same representation. I propose to replace `\n` by a weird unicode symbol that is unlikely to happen, which reduces the interaction space here.

Reply:
How about "♨0A" 
http://www.fileformat.info/info/unicode/char/2668/index.htm plus the hex code for `\n`.

Reply:
@jvican  @eed3si9n  Now `\n` is replaced with `\u2668` as @eed3si9n suggested